xwiki-platform-core/xwiki-platform-component/xwiki-platform-component-wiki/src/main/java/org/xwiki/component/wiki/internal/DefaultWikiObjectComponentManagerEventListener.java

```diff
@@ -87,11 +94,13 @@ public void onEvent(Event event, Object source, Object data)
             EntityReference xObject = xObjectEvent.getReference();
 
             // If the modified XObject can produce a WikiComponent
-            if (this.wikiObjectComponentManagerEventListenerProxy.getWikiObjectsList()
-                    .contains(((BaseObjectReference) xObject).getXClassReference())) {
+            WikiObjectComponentBuilder componentBuilder =
+                    this.getAssociatedComponentBuilder(((BaseObjectReference) xObject).getXClassReference());
```

Comment:
Do you mean that it would be better to get the BaseObject (or the BaseObjectReference) directly by using old core features (such as the XWikiContext) ?

I’m just asking, as this would mean moving this listener to the bridge, and having a major part of a feature implementation depending a lot on the old core might be a bit dangerous.